### PR TITLE
allow for port 0 in listen_interfaces and logs/refactor

### PR DESCRIPTION
**Pull request by aldenml**
two commits here.

any idea of what's happening with appveyor?

### CODE REVIEW

**Comment by arvidn** [comment]
no idea about appveyor. their status page says everything is working. they did report elevated error rates from github on the 18th though, perhaps they're still catching up on the backlog they caused.

File: src/session_impl.cpp [hunk 1]
```diff
@@ -1652,7 +1660,8 @@ namespace aux {
 			}
 
 			if (ec == error_code(error::address_in_use)
-				&& !(flags & listen_no_system_port))
+				&& !(flags & listen_no_system_port)
+				&& bind_ep.port() != 0)
```
Comment (arvidn) [inline src/session_impl.cpp]: this is the key part of the patch, right?

File: src/string_util.cpp [hunk 2]
```diff
@@ -279,7 +278,7 @@ namespace libtorrent
 			else
 			{
 				iface.port = std::atoi(port.c_str());
-				if (iface.port <= 0 || iface.port > 65535) iface.port = -1;
+				if (iface.port < 0 || iface.port > 65535) iface.port = -1;
```
Comment (arvidn) [inline src/string_util.cpp]: and here I suppose

**Comment by arvidn** [comment]
I can't think of why this shouldn't be allowed, and it definitely used to be intended to be allowed.

File: src/string_util.cpp (hunk 2 above)
Comment (aldenml) [inline src/string_util.cpp]: I separated the commits to easy the review (but it's trivial), it's that the PR queue is getting bigger

**Comment by aldenml** [comment]
@arvidn I think you should cancel all the appveyor works, I can trigger new ones in my account

**Comment by arvidn** [comment]
your account works?

**Comment by aldenml** [comment]
without problems

**Comment by aldenml** [comment]
maybe cancelling all the jobs first do the trick

**Pull request merged**